common/courses/participants.ts

```diff
@@ -117,9 +118,39 @@ export function canJoinSubcourses(pupil: Pupil): Decision<CourseDecision> {
     return { allowed: true };
 }
 
+export async function canJoinSubcourse(subcourse: Subcourse, pupil: Pupil): Promise<Decision<CourseDecision>> {
+    if (!canJoinSubcourses(pupil).allowed) {
+        return canJoinSubcourses(pupil);
+    }
+
+    const firstLecture = await prisma.lecture.findMany({
+        where: { subcourseId: subcourse.id },
+        orderBy: { start: 'asc' },
+        take: 1,
+    });
+    if (firstLecture.length !== 1) {
+        return { allowed: false, reason: 'no-lectures' };
+    }
+    if (firstLecture[0].start < new Date() && !subcourse.joinAfterStart) {
+        return { allowed: false, reason: 'already-started' };
+    }
+
+    const pupilGrade = gradeAsInt(pupil.grade);
+    if (subcourse.minGrade && subcourse.maxGrade && subcourse.minGrade < pupilGrade && pupilGrade < subcourse.maxGrade) {
```

Comment:
This line is flawed. I guess

if (subcourse.minGrade && subcourse.minGrade < grade) 
  return { allowed: false };

if (subcourse.maxGrade && subcourse.maxGrade > grade)
  return { allowed: false };

is easier to reason about.

Reply:
ok